site/content/cluster/kubernetes_cluster.md

```diff
@@ -0,0 +1,95 @@
+---
+linktitle: "Kubernetes"
+weight: 1
+---
+
+# Creating an M3 Cluster with Kubernetes
+
+This guide shows you how to create an M3 cluster of 3 nodes, designed to run locally on the same machine. It is designed to show you how M3 and Kubernetes can work together, but not as a production example.
+
+{{% notice note %}}
+This guide assumes you have read the [quickstart](/docs/quickstart), and builds upon the concepts in that guide.
+{{% /notice %}}
+
+{{% notice tip %}}
+We recommend you use [our Kubernetes operator](https://operator.m3db.io/) to deploy M3 to a cluster. It is a more streamlined setup that uses [custom resource definitions](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) to automatically handle operations such as managing cluster placements.
+{{% /notice %}}
+
+
+## Prerequisites
+
+-   A running Kubernetes cluster.
+    -   For local testing, you can use [minikube](https://kubernetes.io/docs/tasks/tools/install-minikube/), [Docker desktop](https://www.docker.com/products/docker-desktop), or [we have a script](https://raw.githubusercontent.com/m3db/m3db-operator/master/scripts/kind-create-cluster.sh) you can use to start a 3 node cluster with [Kind](https://kind.sigs.k8s.io/docs/user/quick-start/).
+-   To use the M3DB operator chart, you need [Helm](https://helm.sh/)
+
+## Create An etcd Cluster
+
+M3 stores its cluster placements and runtime metadata in [etcd](https://etcd.io) and needs a running cluster to communicate with.
+
+We have example services and stateful sets you can use, but feel free to use your own configuration and change any later instructions accordingly.
+
+```shell
+kubectl apply -f https://github.com/m3db/m3db-operator/blob/v0.9.0/example/etcd/etcd-minikube.yaml
+```
+
+If the etcd cluster is running on your local machine, update your _/etc/hosts_ file to match the domains specified in the `etcd` `--initial-cluster` argument. For example to match the `StatefulSet` declaration in the _etcd-minikube.yaml_ above, that is:
+
+```text
+$(minikube ip) etcd-0.etcd
+$(minikube ip) etcd-1.etcd
+$(minikube ip) etcd-2.etcd
+```
+
+Verify that the cluster is running with something like the Kubernetes dashboard, or the command below:
+
+```shell
+kubectl exec etcd-0 -- env ETCDCTL_API=3 etcdctl endpoint health
+```
+
+## Install the Operator with Helm
+
+Add the M3 operator repository:
+
+```shell
+helm repo add m3db https://m3-helm-charts.storage.googleapis.com/stable
```

Comment:
We should test these work, not 100% sure it still does.

Or instead, maybe we just do not offer helm instructions, perhaps we just mention that folks should install the operator by applying the bundle in the m3db-operator repo (much more resilient/always up to date).

Reply:
Ahh, I wasn't aware of that, and certain sections of the current operator docs even imply that helm is the better option. I am happy to remove a dependency, but good to know for certain…

Reply:
But helm certainly still does work…

Reply:
We do update the helm chart on releases. I've seen people mention using it in issues / Slack before so it's definitely used.

That being said, I would say people should default to using the bundle and only use Helm if it's their preferred way of deploying. It's definitely better tested via the bundle.